pkg/auth/authenticator/request/x509request/x509.go

```diff
@@ -92,6 +101,18 @@ func (a *Verifier) AuthenticateRequest(req *http.Request) (user.Info, bool, erro
 	return nil, false, kerrors.NewAggregate(errlist)
 }
 
+func (a *Verifier) verifySubject(subject pkix.Name) error {
+	// No CN restrictions
+	if len(a.allowedCommonNames) == 0 {
+		return nil
+	}
+	// Enforce CN restrictions
+	if a.allowedCommonNames.Has(subject.CommonName) {
+		return nil
+	}
+	return fmt.Errorf("x509: subject with cn=%s is not allowed", subject.CommonName)
```

Comment:
I'm going to assume that you're intentionally withholding the allowed set on the assumption that while a compromised signer means you're owned, they wouldn't be able to also trick openshift without either access to the master-config or the client cert from the existing proxy?

Seems a little thin to me, but if you want it that way, you should add a comment and maybe indicate where in the master-config they should look when they're rejected?